### ISSUE TITLE
SyncBatchNorm raises exception when affine=False

### ISSUE THREAD

**Issue by qasfb** [pytorch/pytorch#54495]
## 🐛 Bug

In nn.SyncBatchNorm:

There is a call to weight.contiguous() there:
https://github.com/pytorch/pytorch/blob/27048c1dfa80effabf17b8dca66cd2724dd502f8/torch/nn/modules/_functions.py#L12

But weight can be None if affine = False.
https://github.com/pytorch/pytorch/blob/27048c1dfa80effabf17b8dca66cd2724dd502f8/torch/nn/modules/batchnorm.py#L44 

So it crashes.

Bug was introduced in:
https://github.com/pytorch/pytorch/commit/d30f4d1dfd5237d89834363ce2cff9de4ee92811

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @albanD @mruberry

**Comment by xwang233** [comment]
Yes, https://github.com/pytorch/pytorch/pull/46906 is related. Thanks for the report.

cc @ngimel
